src/frontend/flight/sql.rs

```diff
@@ -211,27 +212,30 @@ impl FlightSqlService for SeafowlFlightHandler {
                 return Err(Status::invalid_argument(err));
             }
 
-            Some(
-                SyncSchema::try_new(
-                    cmd.column_descriptors.clone(),
-                    batches.first().unwrap().schema(),
-                )
-                .map_err(|err| {
-                    warn!("{err}");
-                    Status::invalid_argument(err.to_string())
-                })?,
+            SyncSchema::try_new(
+                cmd.column_descriptors.clone(),
+                batches.first().unwrap().schema(),
             )
+            .map_err(|err| {
+                warn!("{err}");
+                Status::invalid_argument(err.to_string())
+            })?
         } else {
-            None
+            SyncSchema::empty()
```

Comment:
Can this actually legitimately happen now or should we always expect that there is a schema?

Reply:
It seems that providing an empty payload (`vec![]`) is easier than providing a non-empty payload with empty record batches (`vec![RecordBatch::new_empty(schema)]`), so I think yes.

Let's hear from @eatonphil though.

Reply:
> It seems that providing an empty payload (vec![]) is easier than providing a non-empty payload with empty record batches (vec![RecordBatch::new_empty(schema)]), so I think yes.

Well, let me try. I have such a hard time with the Arrow APIs I was scared to try. But if you think that's better I'll give it a shot.

Reply:
It may still better of to have special casing of the empty payload, than to rely on the rest of the mechanism to adequately process those messages (so this change is for the better anyway).

Reply:
Alright, let's go with it then.